### ISSUE TITLE
Multiboot

### ISSUE THREAD

**Issue by Guizmos** [hieplpvip/ASUS-ZENBOOK-UX410-HACKINTOSH#98]
Hello,

Everything work great (I've UX410 and UX360) except one thing! The multi-boot

I want dual boot in my machine, macOS and Windows 10
I can't boot into Windows because I've blue screen of dead with **acpi_bios_error**

How can I resolve this issue ?

Thanks
(sorry for my bad English)

**Comment by MichaelCasaDev** [comment]
Boot Windows with the Bios BOOTLOADER, using OC (OpenCore) need to edit and recompile the ACPI Patch and adapt them to enable only on "Darwin" (MacOS Kernel) and not into the Win10 system...
But also OC inject all info like SMBIOS into windows, so use OC to boot Windows will be a sh*t!

**Comment by Ubsefor** [comment]
> Boot Windows with the Bios BOOTLOADER, using OC (OpenCore) need to edit and recompile the ACPI Patch and adapt them to enable only on "Darwin" (MacOS Kernel) and not into the Win10 system...
> But also OC inject all info like SMBIOS into windows, so use OC to boot Windows will be a sh*t!

Moreover, even if you do manage to patch DSDT with if-blocks for injecting methods/patches for darwin only, the touchpad won’t work from windows because it needs OSI methods and they are only correctly patched for use in Mac OS (these methods are os-specific – it currently uses some windows OSes to enable pretty good touchpad support, which won’t be available in Windows, sadly). You also need to inject full DSDT of your machine, surrounded by if-block, excluding this injection for darwin, otherwise, Mac OS won’t be happy about it. There are, also, keyboard backlight problems in windows and other issues, some of which apparently can be fixed by installing appropriate drivers from asus.com. But in case @Guizmos wonder if it is possible to use BootCamp from windows – it is not, as Windows doesn’t detect Mac OS partition for some reason. However, you can easily install windows using BootCamp from Mac OS.

TL;DR: It is better to save the hassle and just use bios boot menu to boot windows, if you are up to dual booting.

**Comment by Guizmos** [comment]
Thank you for your answers.
I will therefore use the bios to boot under Windows.
Especially since macOS works wonderfully with this installation.
So simply thank you for your work

**Comment by sj64** [comment]
Bit late to the party, but in case anyone else is looking into this - I use refind: https://www.rodsbooks.com/refind/
Small learning curve but very versatile. I use this to quad boot catalina 10.15.7 (OC), win10 (BCD), elementaryOs Juno, and BlissOs Android (both native, no grub), as well as access UEFI shell, bios, reorder boot preferences etc. all from one custom graphic boot screen.

It basically hands the boot process over natively to whichever os you choose - so no tortuous ACPI edits, Darwin, Bootcamp, nada. Only had to add the ext4 fs driver to recognise linux partitions. It's basically just what UEFI was supposed to be! 

PS I personally find it a blessing Windoze can't read either my mac or linux partitions - it's way too keen on 'fixing' them if it sees them!

![image](https://user-images.githubusercontent.com/64731653/113518280-981a9500-9585-11eb-9ae0-a7846d9b5bc8.png)

**Comment by pkos98** [comment]
@sj64

Congrats!  
I am curious, how did you integrate refind with hackintosh?
I am considering hackintosh, but only in a dual-boot setup with my linux.

**Comment by sj64** [comment]
ReFind detects the opencore installation in EFI automatically, and creates a boot menu item for it. I just tweaked the OC options to hide the picker, set macos as default os, and minimise the wait time to zero, so the handover is invisible, seemingly seamless, and straight to MacOS, but there's no reason why you can't keep the opencore boot menu available.

For Linux you will have to add the ext4_x64.efi driver in the drivers_x64 folder of your refind installation in the efi partition, and specify paths to root, initrd, kernel etc (similar to grub format) to use the built-in linux EFI stub loader in custom menuentry section(s) in refind.conf.
Steve James
Somewhere up a mountain
https://envisioneer.net ***@***.***/0?redirect=https%3A%2F%2Fenvisioneer.net&recipient=cmVwbHkrQVBOM1VCSkpOMjNMNkJRVTRTSzRYTjU2VDZZMlRFVkJOSEhDMk5MN0tZQHJlcGx5LmdpdGh1Yi5jb20%3D)
